src/WebJobs.Extensions.DurableTask/ContextImplementations/DurableOrchestrationClient.cs

```diff
@@ -360,9 +360,20 @@ private async Task<ActorStateResponse<T>> ReadActorState<T>(TaskHubClient client
                 && state.OrchestrationInstance != null
                 && state.Input != null)
             {
-                var serializedState = state.Input;
+                string serializedState;
 
-                var schedulerState = JsonConvert.DeserializeObject<SchedulerState>(state.Input, MessagePayloadDataConverter.MessageSettings);
+                if (this.client.ServiceClient is AzureStorageOrchestrationService service
+                    && state.Input.StartsWith("http"))
+                {
+                    // the input was compressed... read it back from blob
+                    serializedState = await service.DownloadLargeStatusField(state.Input);
```

Comment:
I had some feedback in your other PR that this API should be suffixed with "Async".